lib/s3middleware/objectRestore.ts

```diff
@@ -0,0 +1,141 @@
+import { parseString } from 'xml2js';
+import errors, { ArsenalError } from '../errors';
+import * as werelogs from 'werelogs';
+
+/*
+    Format of xml request:
+    <RestoreRequest>
+      <Days>integer</Days>
+      <Tier>Standard|Bulk|Expedited</RetainUntilDate>
+    </RestoreRequest>
+*/
+
+/**
+ * validateDays - validate restore days
+ * @param days - parsed days string
+ * @return - contains days or error
+ */
+function validateDays(days?: string) {
+    if (!days || !days[0]) {
+        const desc = 'request xml does not contain RestoreRequest.Days';
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    // RestoreRequest.Days must be greater than or equal to 1
+    const daysValue = Number.parseInt(days[0], 10);
+    if (Number.isNaN(daysValue)) {
+        const desc = `RestoreRequest.Days is invalid type. [${days}]`;
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    if (daysValue < 1) {
+        const desc = `RestoreRequest.Days must be greater than 0. [${days}]`;
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    if (daysValue > 2147483647) {
+        const desc = `RestoreRequest.Days must be less than 2147483648. [${days}]`;
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    return { days: daysValue }
+}
+
+/**
+ * validateTier - validate retain until date
+ * @param tier - parsed tier string
+ * @return - contains retain until date or error
+ */
+function validateTier(tier?: string) {
+    // If Tier is specified, must be "Expedited" or "Standard" or "Bulk"
+    const tierList = ['Expedited', 'Standard', 'Bulk'];
```

Comment:
Use a set if we just need to check includes ( and probably move it out of the function scope )

Reply:
yeah, make sense, put in in `constants` file